curriculum/challenges/english/20-upcoming-python/learn-recursion-by-solving-the-tower-of-hanoi-puzzle/64ddd65848a12919d7e1c7d0.md

```diff
@@ -0,0 +1,64 @@
+---
+id: 64ddd65848a12919d7e1c7d0
+title: Step 29
+challengeType: 20
+dashedName: step-29
+---
+
+# --description--
+
+It's time to move some code from the `move` function to the `make_allowed_move` function. Move the code nested inside the outermost `if`, except the first `print` statement, inside your new function. Pay close attention to the indentation.
+
+# --hints--
+
+Test 1
```

Comment:
A try for hint and it's test case:

# --hints--

Keep in mind the indentation and extra spaces.
```js
({ test: () => {
  assert.match(code, /^(\s*)def\s+make_allowed_move\(rod1, rod2\):\n\1\s+forward\s+=\s+False\n\1\s+if\s+len\(rods\[target\]\)\s*==\s*0:/);
  assert.match(code, /^\s+forward\s=\sTrue$|^\s+elif\slen\(rods\[[^\]]+\]\)\s!=\s0\s+and\s+len\(rods\[[^\]]+\]\)\s!=\s0\s+and\srods\[[^\]]+\]\[-1\]\s<\srods\[[^\]]+\]\[-1\]:$/);
  assert.match(code, /^(?:[ \t]*\w+[ \t]*=[ \t]*\w+)?[ \t]*\n(?:[ \t]*if[ \t]+\w+[ \t]*:[ \t]*\n(?:[ \t]*print\([^)]*\)[ \t]*\n[ \t]*\w+\[[\w\s]+\]\.append\(\w+\[[\w\s]+\]\.pop\(\)\)))?/);
  assert.match(code, /^\s*else:\n\s*print\(f'Moving disk \{rods\[target\]\[-1\]\} from \{target\} to \{source\}'\)\n\s*rods\[source\].append\(rods\[target\].pop\(\)\)/);
  assert.match(code, /^(\s*\n)?\s*# display our progress\n\s*print\(rods\)/);  
  }
})
```

Reply:
This step is quite annoying to test... I tried several times unsuccessfully.

Reply:
These aren't working quite right, I'll have another look.